doc/tutorials/snippet-sharing-rest-service.md

```diff
@@ -0,0 +1,196 @@
+```
+kdb set system/sw/elektra/restbackend/#0/current/backend/jwt/encryption_key "use_a_secret_key_here!"
+kdb set system/sw/elektra/restbackend/#0/current/backend/api_specification/html "http://link.to/the/html/version/of/api/description"
+kdb set system/sw/elektra/restbackend/#0/current/backend/api_specification/raw "http://link.to/the/blueprint/of/api/description"
+```
+
+In case you don't want to publish the API description, you can also set the keys
+to other links, e.g. your main page. It is good practice to have the description
+published though.
+
+To change the port of the server to 12345, you can use:
+`kdb set system/sw/elektra/restbackend/#0/current/cppcms/service/port 12345`
+
+### Frontend ###
+
+The frontend does only require small mandatory changes, which have to be made in the
+`application-config.json` in the `/usr/local/share/elektra/tool_data/rest-frontend`
+directory:
+
+- Change `backend.root` to the URL where the backend will be reachable,
+  e.g. `http://restapi.libelektra.org/` (with trailing slash!)
+- Change `website.url` to the URL where the frontend will be reachable,
+  e.g. `http://libelektra.org/` (with trailing slash!)
+
+## Running the Applications ##
+
+As last step we need to run the applications:
+
+- First we start the backend server with `kdb run-rest-backend`.
+  To ensure the backend is accessible, you can use
+  `curl http://localhost:8080/version` (change port to your setting),
+  which should show you some version information in JSON format.
+- Although the frontend was compiled during installation already, we want to
+  have a freshly built homepage and use `kdb configure-rest-frontend` to do so.
+- Then we runt he frontend analogously with `kdb run-rest-frontend`. It should
+  now be reachable at the configured port.
+
+If everything went smooth, both applications should now be online and reachable.
+
+## Stopping the Appplications ##
+
+Both applications can be stopped with a simple command:
+
+- Backend: `kdb stop-rest-backend`
+- Frontend: `kdb stop-rest-frontend`
+
+## Additional Tasks ##
+
+### API Specification ###
+
+The API of the backend was designed with an API blueprint very detailed.
+To compile the [blueprint](/doc/api_blueprints/snippet-sharing.md) we need
+the [apiary-client](https://github.com/apiaryio/apiary-client) to be installed,
+which in return is installed as Ruby gem:
+
+- So we first install Ruby with gem: `apt-get install ruby`
+- Then we install the apiary-cli: `gem install apiaryio`
+- Finally we can use `apiary preview --path=snippet-sharing.apib --output=snippet-sharing-api.html`
+  in the [blueprints](/doc/api_blueprints) directory to generate the pretty version
+  of the API description. This file can then be placed wherever it is accessible
+  or needed.
+
+In case you change the API description (and the backend), you may want to ensure
+that your API blueprint is still syntax conform. To do so, you can use the tool
+[Drafter](https://github.com/apiaryio/drafter).
```

Comment:
I think that is out of scope for the tutorial and it is described quite well on their github page (therefore the link). When you have it installed, you simply run `drafter <file>`.